### DISCUSSION TITLE
How to form intersections

### DISCUSSION THREAD

**Discussion by kmamor** [gama-platform/gama#207] [Q&A]
In the model 'drive random' (here [https://github.com/gama-platform/gama.old/blob/GAMA_1.9.3/simtools.gaml.extensions.traffic/models/Driving%20Skill/models/Advanced%20models/Drive%20Random.gaml]  intersections are formed from a shapefile called nodes.shp (an example name) presumably a shapefile comprising nodes. Looking  at OSM via QGIS I cannot see how to create a file of road intersections comprising nodes. It is possible to create a layer just comprising traffic lights  - is that what is meant here? 

Thanks,

K

**Comment by kmamor** [comment]
To illustrate the problem here is a screen grab. 
![error3](https://github.com/gama-platform/gama/assets/45757038/05c4fb91-d7ec-48b1-b5a0-feb4e4cefa7c)


In QGIS I selected motorway/trunk/primary/secondary and tertiary highways together with their links. Under advanced I only clicked 'nodes' but got a message saying  there was no output. So I clicked nodes and points and got the illustrated layer. As can be seen the nodes are sometimes on intersections but lots of intersections do not have a node,  Often the nodes are not on intersections at all. 

What I am trying to achieve is to experiment with the driving skill. In the example code it refers to a node.shp. It seems to me that GAMA cannot detect intersections directly from a road.shp. Thus, my understanding is that I need to convert the intersections into nodes and save a shapefile from a package such as QGIS. 

Any help appreciated. Is there somewhere I should consider posting queries? I had a look on stack overload but there does not seem to be GAMA community. 

Thanks,


K

**Reply by kmamor** [reply]
I have managed to clean up the layer somewhat (using dissolve etc) and now have this:

![error5](https://github.com/gama-platform/gama/assets/45757038/4b7f05e0-b157-4117-b46c-4715ed34b447)

This does seem to be a layer of just nodes at intersections. I have saved the nodes as a shapefile to the includes folder and now get an error when I try to run the model. The error message references this file. 

![error6](https://github.com/gama-platform/gama/assets/45757038/eb9fa987-e10f-4ad6-a109-8646d3d2f7df)


Any help very much appreciated - or if you could please direct me to a community (like stack overload )

Thanks

K

**Reply by lesquoyb** [reply]
my guess is that your file is not really in the includes folder but you have a link. Could you open that folder and check if it's really there ?
Otherwise please copy the error message and send you shp file
